### PR TITLE
Repro of issue 5560

### LINKED ISSUE #5560: [Android] Disposed EntryCell throws ObjectDisposed exception after updating an object that the EntryCell was previously bound to
### Description

Disposed EntryCell throws ObjectDisposed exception after updating an object that the EntryCell was previously bound to. This doesnt happen to plain Entry or Editor.

### Steps to Reproduce

1. create a (shared) bindable object
2. bind it to an entrycell.text
3. release the tabview with the entrycell (probably wont be cleaned up properly at this point)
4. update the bindable object -> ObjectDisposed exception

this issue may be related to #5555

### Expected Behavior

no exception (no dangling binding?)

### Actual Behavior

System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'Xamarin.Forms.Platform.Android.EntryCellEditText'.
  at Java.Interop.JniPeerMembers.AssertSelf (Java.Interop.IJavaPeerable self) [0x00029] in <286213b9e14c442ba8d8d94cc9dbec8e>:0 
  at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeVirtualObjectMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x00000] in <286213b9e14c442ba8d8d94cc9dbec8e>:0 
  at Android.Widget.TextView.get_TextFormatted () [0x0000a] in <b781ed64f1d743e7881ac038e0fbdf85>:0 
  at Android.Widget.TextView.get_Text () [0x00000] in <b781ed64f1d743e7881ac038e0fbdf85>:0 
  at Xamarin.Forms.Platform.Android.EntryCellRenderer.UpdateText () [0x0000c] in D:\a\1\s\Xamarin.Forms.Platform.Android\Cells\EntryCellRenderer.cs:140 
  at Xamarin.Forms.Platform.Android.EntryCellRenderer.OnCellPropertyChanged (System.Object sender, System.ComponentModel.PropertyChangedEventArgs e) [0x0003d] in D:\a\1\s\Xamarin.Forms.Platform.Android\Cells\EntryCellRenderer.cs:47 
  at Xamarin.Forms.Platform.Android.CellRenderer.OnGlobalCellPropertyChanged (System.Object sender, System.ComponentModel.PropertyChangedEventArgs e) [0x0001d] in D:\a\1\s\Xamarin.Forms.Platform.Android\Cells\CellRenderer.cs:130 
  at Xamarin.Forms.BindableObject.OnPropertyChanged (System.String propertyName) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:173 
  at Xamarin.Forms.Element.OnPropertyChanged (System.String propertyName) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Element.cs:339 
  at Xamarin.Forms.BindableObject.SetValueActual (Xamarin.Forms.BindableProperty property, Xamarin.Forms.BindableObject+BindablePropertyContext context, System.Object value, System.Boolean currentlyApplying, Xamarin.Forms.Internals.SetValueFlags attributes, System.Boolean silent) [0x00114] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:622 
  at Xamarin.Forms.BindableObject.SetValueCore (Xamarin.Forms.BindableProperty property, System.Object value, Xamarin.Forms.Internals.SetValueFlags attributes, Xamarin.Forms.BindableObject+SetValuePrivateFlags privateAttributes) [0x00179] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:422 
  at Xamarin.Forms.BindingExpression.ApplyCore (System.Object sourceObject, Xamarin.Forms.BindableObject target, Xamarin.Forms.BindableProperty property, System.Boolean fromTarget) [0x00220] in D:\a\1\s\Xamarin.Forms.Core\BindingExpression.cs:167 
  at Xamarin.Forms.BindingExpression.Apply (System.Boolean fromTarget) [0x0003e] in D:\a\1\s\Xamarin.Forms.Core\BindingExpression.cs:57 
  at Xamarin.Forms.BindingExpression+BindingExpressionPart.<PropertyChanged>b__49_0 () [0x00000] in D:\a\1\s\Xamarin.Forms.Core\BindingExpression.cs:618 
  at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <b781ed64f1d743e7881ac038e0fbdf85>:0 
  at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in <b781ed64f1d743e7881ac038e0fbdf85>:0 
  at (wrapper dynamic-method) System.Object.34(intptr,intptr)

### Basic Information

- Version with issue: 3.6.0.
- IDE: VS 15.9.8
- Platform Target Frameworks:
  - Android: 9.0

### Reproduction Link

with the attached test:
[FormsTest.zip](https://github.com/xamarin/Xamarin.Forms/files/2966689/FormsTest.zip)

1. click on open leaking page
2. edit the entrycell
3. navigate back
4. modify the entry
5. click GC
6. try modifying the entry again (should crash)

### PR DESCRIPTION
**Pull request by YoupHulsebos**
### Description of Change ###
Added repro for this issue

### Issues Resolved ### 
 - related to #5560 

### API Changes ###
None
 
### Platforms Affected ### 
- Android

### Behavioral/Visual Changes ###
None

### Before/After Screenshots ### 
Not applicable

### Testing Procedure ###
 - Find the testcase
 - Open the leak page
 - Edit text
 - Go back
 - Press GC collect button
 - Edit text

### PR Checklist ###
- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by samhouts** [comment]
It is actually reproducible!

```csharp
[Preserve(AllMembers = true)]
	[Issue(IssueTracker.Github, 5560, "[Android] Disposed EntryCell throws ObjectDisposed exception after updating an object that the EntryCell was previously bound to", PlatformAffected.Android)]
	public class Issue5560 : TestNavigationPage
	{
		protected override void Init()
		{
			Navigation.PushAsync(new NavigationPage(new Page1()));
		}

		[Preserve(AllMembers = true)]
		class Page1 : ContentPage
		{
			public Entry SharedObject { get; } = new Entry { Text = "test" };

			public Page1()
			{
				var button1 = new Button { Text= "Open leaking page" };
				button1.Clicked += Button1_Clicked;

				var button2 = new Button { Text = "GC" };
				button2.Clicked += Button2_Clicked;

				var entry = new Entry();
				entry.SetBinding(Entry.TextProperty, new Binding("Text", source: SharedObject));

				Content = new StackLayout
				{
					Children = {
						button1,
						button2,
						entry
					}
				};
			}

			void Button2_Clicked(object sender, EventArgs e)
			{
				GC.Collect();
				GC.WaitForPendingFinalizers();
				GC.Collect();
			}

			void Button1_Clicked(object sender, EventArgs e)
			{
				Navigation.PushAsync(new Page2(SharedObject));
			}
		}

		[Preserve(AllMembers = true)]
		class Page2 : ContentPage
		{
			public object SharedObject { get; }

			public Page2(object sharedObject)
			{
				SharedObject = sharedObject;

				var item = new EntryCell { Label = "i leak" };
				item.SetBinding(EntryCell.TextProperty, new Binding("Text", source: SharedObject));

				var tableView = new TableView { 
					Root = new TableRoot
					{
						new TableSection
						{
							item
						}
					}
				};

				Content = new StackLayout
				{
					Children = {
						tableView
					}
				};
			}
		}

	}
```

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue5560.cs [hunk 1]
```diff
@@ -86,5 +94,24 @@ public Page2(object sharedObject)
...
+		{
+			RunningApp.WaitForElement(q => q.Marked("NoResourceEntry-112"));
+			RunningApp.Tap(q => q.Marked("NoResourceEntry-112"));
+			RunningApp.Tap(c => c.Class("EntryCellEditText"));
+			RunningApp.EnterText(c => c.Class("EntryCellEditText"), " edit");
+			RunningApp.Back();
+			RunningApp.Back();
+			RunningApp.Tap(q => q.Marked("NoResourceEntry-113"));
+			RunningApp.Tap(q => q.Marked("NoResourceEntry-114"));
+			RunningApp.EnterText(c => c.Marked("NoResourceEntry-114"), " twice");
```
Comment (YoupHulsebos) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue5560.cs]: This doens't seem to reproduce it. If I do the same manually it does work

**Comment by jfversluis** [comment]
What is the status of this? I'm a bit confused... :D 

@YoupHulsebos with the code from @samhouts you still find that you cannot reproduce this?

**Comment by YoupHulsebos** [comment]
I was able to reproduce the issue manually. I wasn't able to reproduce with an automated UITest. I was still working on this. And it is definitely not ready to merge.

**Comment by jfversluis** [comment]
Noted and marked as such! :)

Please let us know if you could use any assistance of when you've made any progress.

**Pull request closed without merging**